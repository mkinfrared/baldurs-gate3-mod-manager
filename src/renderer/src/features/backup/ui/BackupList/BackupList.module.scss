.BackupList {
  background-color: orange;
  max-width: 60rem;
}
