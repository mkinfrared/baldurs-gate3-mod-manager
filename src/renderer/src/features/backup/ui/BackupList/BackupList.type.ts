import { Game } from "@renderer/shared/types";

export type BackupListProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  game: Game;
};
