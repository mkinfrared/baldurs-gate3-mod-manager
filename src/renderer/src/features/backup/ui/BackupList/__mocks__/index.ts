export { BackupListMock as BackupList } from "./BackupList.mock";
