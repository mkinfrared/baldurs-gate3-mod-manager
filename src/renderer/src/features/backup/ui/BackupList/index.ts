export { BackupList } from "./BackupList";

export * from "./BackupList.type";
