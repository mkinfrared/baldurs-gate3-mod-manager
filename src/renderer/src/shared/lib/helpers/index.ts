export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { getNodeText } from "./getNodeText";

export { lazyImport } from "./lazyImport";

export { mergeRefs } from "./mergeRefs";

export { trpcClient, trpc } from "./trpc";

export * from "./typeGuards";
