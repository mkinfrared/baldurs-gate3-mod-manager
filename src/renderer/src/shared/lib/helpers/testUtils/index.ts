export { renderWithRouter, withQueryProvider } from "./testUtils";
