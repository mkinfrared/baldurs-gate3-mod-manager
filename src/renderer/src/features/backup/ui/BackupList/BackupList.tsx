import { classNames, trpc } from "@renderer/shared/lib/helpers";

import { Backup } from "../Backup";

import css from "./BackupList.module.scss";
import { BackupListProps } from "./BackupList.type";

const BackupList = ({ className, game }: BackupListProps) => {
  const { data } = trpc.backup.getBackups.useQuery(game);

  return (
    <div
      className={classNames(css.BackupList, className)}
      data-testid="BackupList"
    >
      {data?.map(({ filePath, name }) => (
        <Backup key={filePath} name={name} filePath={filePath} game={game} />
      ))}
    </div>
  );
};

export { BackupList };
