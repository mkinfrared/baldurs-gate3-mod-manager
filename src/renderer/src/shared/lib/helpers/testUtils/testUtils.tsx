/* eslint-disable import/no-unused-modules */
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { ReactElement, ReactNode } from "react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

import { trpc, trpcClient } from "@renderer/shared/lib/helpers/trpc";

// https://testing-library.com/docs/example-react-router/
const renderWithRouter = (ui: ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

const queryClient = new QueryClient();

const withQueryProvider = (children: ReactNode) => (
  <trpc.Provider client={trpcClient} queryClient={queryClient}>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </trpc.Provider>
);

const withRouter = (children: ReactNode) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export { renderWithRouter, withQueryProvider, withRouter };
