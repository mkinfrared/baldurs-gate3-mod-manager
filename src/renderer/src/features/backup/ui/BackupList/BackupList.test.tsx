import { render } from "@testing-library/react";

import { withQueryProvider } from "@renderer/shared/lib/helpers/testUtils";

import { BackupList } from "./BackupList";

describe("<BackupList />", () => {
  const game = "BG3";
  const Component = <BackupList game={game} />;

  it("should be defined", () => {
    expect(BackupList).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(withQueryProvider(Component));

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(withQueryProvider(Component));
    const element = getByTestId("BackupList");

    expect(element).toBeDefined();
  });
});
