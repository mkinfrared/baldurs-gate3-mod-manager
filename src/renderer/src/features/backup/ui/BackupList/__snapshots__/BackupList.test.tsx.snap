// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<BackupList /> > should match the snapshot 1`] = `
<div>
  <div
    class="BackupList"
    data-testid="BackupList"
  />
</div>
`;
